drivers/platform/mbed/mbed_uart.cpp

```diff
@@ -0,0 +1,420 @@
+						     desc->extra))->uart_port);
+		return uart->write(data, bytes_number);
+	}
+}
+
+/**
+ * @brief Submit reading buffer to the UART driver.
+ * Buffer is used until bytes_number bytes are read.
+ * @param desc[in]- Descriptor of the UART device
+ * @param data[out] - Buffer where data will be read
+ * @param bytes_number[in] - Number of bytes to be read.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ * @note Currently implemented only for UART and not for USBCDC (VCOM)
+ */
+int32_t uart_read_nonblocking(struct uart_desc *desc,
+			      uint8_t *data,
+			      uint32_t bytes_number)
+{
+	mbed::UnbufferedSerial *uart;	// pointer to UnbufferedSerial/UART instance
+	uint32_t i;
+
+	if (!desc || !desc->extra || !data)
+		return -EINVAL;
+
+	if (!((struct mbed_uart_desc *)(desc->extra))->uart_port)
+		return -EINVAL;
+
+	uart = (UnbufferedSerial *)(((struct mbed_uart_desc *)(
+					     desc->extra))->uart_port);
+
+	for (i = 0; i < bytes_number; i++) {
+		if (uart->readable() > 0)
+			uart->read(&data[i], 1);
+	}
+
+	return bytes_number;
+}
+
+/**
+ * @brief Submit writting buffer to the UART driver.
+ * Data from the buffer is sent over the UART, the function returns immediatly.
+ * @param desc[in] - Descriptor of the UART device
+ * @param data[in,out] - Buffer where data will be written
+ * @param bytes_number[in] - Number of bytes to be written.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ * @note Currently implemented only for UART and not for USBCDC (VCOM)
+ */
+int32_t uart_write_nonblocking(struct uart_desc *desc,
+			       const uint8_t *data,
+			       uint32_t bytes_number)
+{
+	mbed::UnbufferedSerial *uart;		// pointer to UnbufferedSerial/UART instance
+	uint32_t i;
+
+	if (!desc || !desc->extra || !data)
+		return - EINVAL;
+
+	if (!((struct mbed_uart_desc *)(desc->extra))->uart_port)
+		return -EINVAL;
+
+	uart = (UnbufferedSerial *)(((struct mbed_uart_desc *)(
+					     desc->extra))->uart_port);
+
+	for (i = 0; i < bytes_number; i++)
+		uart->write(&data[i], 1);
+
+	return bytes_number;
+}
+
+/**
+ * @brief Initialize the UART communication peripheral.
+ * @param desc[in, out] - The UART descriptor.
+ * @param param[in] - The structure that contains the UART parameters.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t uart_init(struct uart_desc **desc, struct uart_init_param *param)
+{
+	mbed::UnbufferedSerial *uart;	// Pointer to new UnbufferedSerial/UART instance
+	platform_usbcdc *usb_cdc_dev;	// Pointer to usb cdc device class instance
+	struct mbed_uart_desc *mbed_uart_desc;	// Pointer to mbed uart descriptor
+	struct uart_desc *uart_desc; 			// UART descriptor
+
+	if (!desc || !param)
+		return -EINVAL;
+
+	// Create the UART description object for the device
+	uart_desc = (struct uart_desc *)calloc(1, sizeof(*uart_desc));
+	if (!uart_desc)
+		return -ENOMEM;
+
+	uart_desc->baud_rate = param->baud_rate;
```

Comment:
I find it interesting that you don't use the uart_init_param fields, except for this one - baud_rate. The unused ones are:
```
/** UART Device ID */
uint8_t	device_id;
/** UART number of data bits */
enum uart_size		size;
/** UART parity */
enum uart_parity	parity;
/** UART number of stop bits */
enum uart_stop		stop;
```

Reply:
Thanks for pointing this. We were always using default values for these parameters however it make sense to update them in the mbed layer for custom needs. I will updated the drivers accordingly.